File: rust/src/models/list_response_message_attempt_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::message_attempt_out::MessageAttemptOut;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ListResponseMessageAttemptOut {
    pub data: Vec<MessageAttemptOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(rename = "prevIterator")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseMessageAttemptOut {
    pub fn new(data: Vec<MessageAttemptOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }
}

File: kotlin/lib/src/main/kotlin/models/ListResponseMessageAttemptOut.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class ListResponseMessageAttemptOut(
    val data: List<MessageAttemptOut>,
    val done: Boolean,
    val iterator: String? = null,
    val prevIterator: String? = null,
)
